pkg/snapstore/snapstore_test.go

```diff
@@ -278,6 +280,220 @@ var _ = Describe("Save, List, Fetch, Delete from mock snapstore", func() {
+							Expect(err).ShouldNot(HaveOccurred())
+							Expect(lastCreationTime.IsZero()).ShouldNot(BeTrue())
+							err = os.Remove(filepath.Join(credentialDirectory, credentialFile))
+							Expect(err).ShouldNot(HaveOccurred())
+							// remove one credential file and then run the test
+							newSecretModifiedTime, err := GetSnapstoreSecretModifiedTime(config.SnapstoreProvider)
+							Expect(err).Should(HaveOccurred())
+							Expect(newSecretModifiedTime.IsZero()).Should(BeTrue())
+						})
+					}
+				})
+				Context("credentials exist", func() {
+					var lastCreationTime time.Time
+					BeforeEach(func() {
+						var err error
+						lastCreationTime, err = createCredentialFilesInDirectory(credentialDirectory, config.CredentialFiles)
+						Expect(err).ShouldNot(HaveOccurred())
+						Expect(lastCreationTime.IsZero()).ShouldNot(BeTrue())
+					})
+					// unmodified credentials
+					It("should return the latest creation time among the credential files", func() {
+						newSecretModifiedTime, err := GetSnapstoreSecretModifiedTime(config.SnapstoreProvider)
+						Expect(err).ShouldNot(HaveOccurred())
+						Expect(newSecretModifiedTime.Equal(lastCreationTime)).Should(BeTrue())
+					})
+					// modified credentials
+					for _, credentialFile := range config.CredentialFiles {
+						credentialFile := credentialFile
+						It(fmt.Sprintf("should return the modification time of the credential file %s", credentialFile), func() {
+							fmt.Println("Testing dynamic credential rotation for the provider:", config.Provider)
+							err := modifyCredentialFileInDirectory(credentialDirectory, credentialFile)
+							Expect(err).ShouldNot(HaveOccurred())
+							newSecretModifiedTime, err := GetSnapstoreSecretModifiedTime(config.SnapstoreProvider)
+							Expect(err).ShouldNot(HaveOccurred())
+							Expect(newSecretModifiedTime.After(lastCreationTime)).Should(BeTrue())
+						})
+					}
+				})
+			})
+		})
+	}
+})
+
+// creates the access credential files in the temporary directory.
+// the function also returns the timestamp at which the last file was created
+func createCredentialFilesInDirectory(directory string, filenames []string) (time.Time, error) {
+	for _, filename := range filenames {
+		filename := filepath.Join(directory, filename)
+		// creates and writes content to the file
+		err := os.WriteFile(filename, []byte("INITIAL CONTENT"), os.ModePerm)
+		if err != nil {
+			return time.Time{}, err
+		}
+	}
+	// the last file created at the end has the latest creation time (modification time)
+	fileName := filepath.Join(directory, filenames[len(filenames)-1])
+	lastFile, err := os.Open(fileName)
+	if err != nil {
+		return time.Time{}, err
+	}
+	defer lastFile.Close()
+	lastFileInfo, err := lastFile.Stat()
+	if err != nil {
+		return time.Time{}, err
+	}
+	return lastFileInfo.ModTime(), nil
+}
+
+// modifies the credential file in the credential directory
```

Comment:
```suggestion
// modifyCredentialFileInDirectory modifies a specific credential file within the given directory.
```

Reply:
Addressed in [8635c10](https://github.com/gardener/etcd-backup-restore/pull/670/commits/8635c101d031f890343da3177cb410aa20053dc5)